provider/cloudflare/cloudflare.go

```diff
@@ -287,21 +291,29 @@ func NewCloudFlareProvider(
 ) (*CloudFlareProvider, error) {
 	// initialize via chosen auth method and returns new API object
 	var (
-		config *cloudflare.API
-		err    error
+		config   *cloudflare.API
+		configV4 *cloudflarev4.Client
+		err      error
 	)
 	if os.Getenv("CF_API_TOKEN") != "" {
 		token := os.Getenv("CF_API_TOKEN")
-		if strings.HasPrefix(token, "file:") {
-			tokenBytes, err := os.ReadFile(strings.TrimPrefix(token, "file:"))
+		if path, ok := strings.CutPrefix(token, "file:"); ok {
```

Comment:
Worth to probably fix/review a bug first https://github.com/kubernetes-sigs/external-dns/issues/5400

As not, we may introduce other issues or actually fix a bug silently

Reply:
Make sense, better to made this change as a standalone PR